src/main/java/net/minecraftforge/network/ForgeNetwork.java

```diff
@@ -0,0 +1,263 @@
+
+import java.util.Set;
+import java.util.stream.Stream;
+
+import io.netty.buffer.Unpooled;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.inventory.EquipmentSlotType;
+import net.minecraft.inventory.container.Slot;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.IPacket;
+import net.minecraft.network.NetworkManager;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.server.ServerChunkProvider;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.common.ForgeMod;
+import net.minecraftforge.fml.network.NetworkDirection;
+import net.minecraftforge.fml.network.NetworkRegistry;
+import net.minecraftforge.fml.network.simple.SimpleChannel;
+import net.minecraftforge.network.message.SyncBlockEntityCapabilities;
+import net.minecraftforge.network.message.SyncEntityCapabilities;
+import net.minecraftforge.network.message.SyncEquipmentSlotCapabilities;
+import net.minecraftforge.network.message.SyncSlotCapabilities;
+
+public class ForgeNetwork {
+
+    public static final String PROTOCOL_VERSION = "0.0.1";
+    public static final ResourceLocation CHANNEL_NAME = new ResourceLocation(ForgeMod.ID, "main");
+    public static final SimpleChannel CHANNEL = NetworkRegistry.newSimpleChannel(CHANNEL_NAME, () -> PROTOCOL_VERSION,
+        NetworkRegistry.acceptMissingOr(PROTOCOL_VERSION), PROTOCOL_VERSION::equals);
+
+    private static boolean initialized = false;
+
+    public static void init()
+    {
+        if (initialized)
+            throw new IllegalStateException("Forge network already initialized");
+
+        CHANNEL.messageBuilder(SyncSlotCapabilities.class, 0x00, NetworkDirection.PLAY_TO_CLIENT)
+            .encoder(SyncSlotCapabilities::encode)
+            .decoder(SyncSlotCapabilities::decode)
+            .consumer(SyncSlotCapabilities::handle)
+            .add();
+
+        CHANNEL.messageBuilder(SyncEntityCapabilities.class, 0x01, NetworkDirection.PLAY_TO_CLIENT)
+            .encoder(SyncEntityCapabilities::encode)
+            .decoder(SyncEntityCapabilities::decode)
+            .consumer(SyncEntityCapabilities::handle)
+            .add();
+
+        CHANNEL.messageBuilder(SyncBlockEntityCapabilities.class, 0x03, NetworkDirection.PLAY_TO_CLIENT)
+            .encoder(SyncBlockEntityCapabilities::encode)
+            .decoder(SyncBlockEntityCapabilities::decode)
+            .consumer(SyncBlockEntityCapabilities::handle)
+            .add();
+
+        CHANNEL.messageBuilder(SyncEquipmentSlotCapabilities.class, 0x04, NetworkDirection.PLAY_TO_CLIENT)
+            .encoder(SyncEquipmentSlotCapabilities::encode)
+            .decoder(SyncEquipmentSlotCapabilities::decode)
+            .consumer(SyncEquipmentSlotCapabilities::handle)
+            .add();
+    }
```

Comment:
This is also a point to discuss, if there is a better solution to do this or not.